patches/server/0738-Hide-unnecessary-itemmeta-from-clients.patch

```diff
@@ -0,0 +1,80 @@
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index 2b7eeb5659b1083ef550eb9feb0b7ba8a92a92e3..4025f156222ff09f3cb6d8c42aef5d9e0a0d845f 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -3041,6 +3041,53 @@ public abstract class LivingEntity extends Entity {
+         equipment.forEach((enumitemslot, itemstack) -> {
+             ItemStack itemstack1 = itemstack.copy();
+ 
++            // Paper start - hide unnecessary itemmeta from clients.
++            if (itemstack1.getTag() != null && this.level.paperConfig.hideItemmetaFromClients) {
++                CompoundTag nbt = itemstack1.getTag();
++                itemstack1.setCount(1);
++                // Only show damage values for elytras.
++                if (itemstack1.is(Items.ELYTRA) && itemstack1.getDamageValue() >= itemstack1.getMaxDamage() - 1) {
++                    itemstack1.setDamageValue(itemstack1.getDamageValue());
++                } else {
++                    itemstack1.setDamageValue(0);
++                }
++
++                // We can't just strip out display, leather helmets still use the display.color tag.
++                if (nbt.get("display") instanceof CompoundTag displayTag) {
++                    displayTag.remove("Lore");
++                    displayTag.remove("Name");
++                }
++
++                if (nbt.get("Enchantments") instanceof ListTag enchantmentsTag && !enchantmentsTag.isEmpty()) {
++                    // The client still renders items with the enchantment glow if the enchantments tag contains at least one (empty) child.
++                    ListTag enchantments = new ListTag();
++                    enchantments.add(new CompoundTag());
++                    nbt.put("Enchantments", enchantments);
++                }
++                nbt.remove("AttributeModifiers");
++
++                // Bundles change their texture based on their fullness.
++                if (itemstack1.is(Items.BUNDLE) && nbt.get("Items") instanceof ListTag oldItems && !oldItems.isEmpty()) {
++                    org.bukkit.inventory.meta.BundleMeta bundleMeta = (org.bukkit.inventory.meta.BundleMeta) itemstack1.asBukkitMirror().getItemMeta();
++                    int sizeUsed = 0;
++                    for (org.bukkit.inventory.ItemStack item : bundleMeta.getItems()) {
++                        int scale = 64 / item.getMaxStackSize();
++                        sizeUsed += scale * item.getAmount();
++                    }
++                    // Now we add a single fake item that uses the same amount of slots as all other items.
++                    ListTag items = new ListTag();
++                    CompoundTag fakeItem = new CompoundTag();
++                    fakeItem.putInt("Count", sizeUsed);
++                    fakeItem.putString("id", "minecraft:paper");
```

Comment:
Hehe, I see what you did here. Either way, you should use the `Items.PAPER` constant for this.